server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionService.java

```diff
@@ -223,15 +224,22 @@ private boolean reconcile(Set<CompactionJob> jobs, Collection<SubmittedJob> subm
     return true;
   }
 
-  public void compact(CompactionKind kind, Compactable compactable,
+  /**
+   * Get compaction plan for the provided compactable tablet and possibly submit for compaction.
+   * Plans get added to the planning queue before calling the planningExecutor to get the plan. If
+   * no files are selected, return. Otherwise, submit the compaction job.
+   */
+  public void submitCompaction(CompactionKind kind, Compactable compactable,
       Consumer<Compactable> completionCallback) {
     Objects.requireNonNull(compactable);
 
+    // add tablet to planning queue and use planningExecutor to get the plan
     if (queuedForPlanning.get(kind).putIfAbsent(compactable.getExtent(), compactable) == null) {
       try {
         planningExecutor.execute(() -> {
           try {
-            planCompaction(kind, compactable, completionCallback);
+            Optional<CompactionPlan> plan = getCompactionPlan(kind, compactable);
+            plan.ifPresent(cp -> submitCompactionJob(cp, compactable, completionCallback));
```

Comment:
As mentioned in other comments, should probably not add candidates to plan and the reason it was added does not actually log all of the info.  Think the following changes may get the needed information where it needs to go w/o modifying the plan interface.

```suggestion
            var files = compactable.getFiles(...);
            Optional<CompactionPlan> plan = getCompactionPlan(kind, files, compactable.getExtent());
            plan.ifPresent(cp -> submitCompactionJob(cp, files, compactable.getExtent(), completionCallback));
```

Reply:
Thanks. I like this better. I thought getting the files was part of getting the plan but it looks like the plan is more about the jobs.